src/communication/xcSession.ts

```diff
@@ -6,140 +6,157 @@ import xcWebSocketBridgeConfiguration from "configuration/xcWebSocketBridgeConfi
+        }, this);
+    };
 
+    removePrivateTopic(privateTopic: string): void {
+        let kindPrivate = xcWebSocketBridgeConfiguration.kinds.Private;
+        this.privateSubscriber.sendUnsubscribeRequestToTopic(privateTopic, kindPrivate);
+        this.removeElement(this.privateTopics, privateTopic);
+        this.subscribers.forEach(function (subscriber) {
+            subscriber.privateTopics = this.privateTopics;
+        }, this);
+    };
 
-Session.prototype.init = function (sessionListener, getXcApiRequest, xcApiFileName) {
-    let thisObject = this;
+    init(openListener: (e: Event) => void, errorListener: (err: Error) => void): void {
+        let thisSession = this;
+
+        this.webSocket.onopen = function (e: Event) {
+            thisSession.privateSubscriber.sendSubscribeRequestToTopic(thisSession.privateTopic, xcWebSocketBridgeConfiguration.kinds.Private);
+            openListener(e);
+            console.log("connection started on " + thisSession.serverUrl + ".");
+        };
+
+        this.webSocket.onerror = function (e: Event) {
+            let messageError = "Error on " + thisSession.serverUrl + ".";
+            console.error(messageError);
+            console.error(e);
+            errorListener(new Error(messageError));
+        };
+
+        this.webSocket.onclose = function (e: CloseEvent) {
+            console.log("connection on " + thisSession.serverUrl + " closed.");
+            console.log(e);
+            if (!e.wasClean) {
+                throw new Error("unexpected connection close with code " + e.code);
+            }
+        };
+    };
 
-    this.webSocket.onopen = function (e) {
-        thisObject.privateSubscriber.sendSubscribeRequestToTopic(thisObject.privateTopic, xcWebSocketBridgeConfiguration.kinds.Private);
-        getXcApiRequest(xcApiFileName, sessionListener);
-        console.log("connection started on " + thisObject.serverUrl + ".");
+    createPublisher(): Publisher {
+        let publisher = new Publisher(this.webSocket, this.configuration, this.privateTopic, this.sessionData);
+        this.publishers.push(publisher);
+        return publisher;
     };
 
-    this.webSocket.onerror = function (e) {
-        let messageError = "Error on " + thisObject.serverUrl + ".";
-        console.error(messageError);
-        console.error(e);
-        sessionListener(messageError, null);
+    createSubscriber(): Subscriber {
+        let subscriber = new Subscriber(this.webSocket, this.configuration, this.replyPublisher, this.guid, this.privateTopics);
+        this.subscribers.push(subscriber);
+        return subscriber;
     };
 
-    this.webSocket.onclose = function (e) {
-        console.log("connection on " + thisObject.serverUrl + " closed.");
-        console.log(e);
-        if (!e.wasClean) {
-            throw new Error("unexpected connection close with code " + e.code);
+    private removeElement<T>(array: Array<T>, e: T): void {
```

Comment:
Create an extension method for Array.
This method is already defined in definition.d.ts